### PR TITLE
[SPIR-V] Fix emission of debug and annotation instructions and add SPV_EXT_optnone SPIR-V extension

### PR DESCRIPTION
**Pull request by VyacheslavLevytskyy**
This PR fixes:
* emission of OpNames (added newly inserted internal intrinsics and basic blocks)
* emission of function attributes (SRet is added)
* implementation of SPV_INTEL_optnone so that it emits OptNoneINTEL Function Control flag, and add implementation of the SPV_EXT_optnone SPIR-V extension.

### CODE REVIEW

**Comment by VyacheslavLevytskyy** [comment]
CC: @MrSidims

**Comment by VyacheslavLevytskyy** [comment]
The only fail in the test cases is CodeGen/SPIRV/debug-info/debug-type-basic.ll that is unrelated to this PR, see https://github.com/llvm/llvm-project/issues/118011

File: llvm/lib/Target/SPIRV/SPIRVModuleAnalysis.cpp [hunk 1]
```diff
@@ -1606,6 +1610,28 @@ static void addDecorations(const Module &M, const SPIRVInstrInfo &TII,
...
 
+static void addMBBNames(const Module &M, const SPIRVInstrInfo &TII,
+                        MachineModuleInfo *MMI, const SPIRVSubtarget &ST,
+                        SPIRV::ModuleAnalysisInfo &MAI) {
+  for (auto F = M.begin(), E = M.end(); F != E; ++F) {
+    MachineFunction *MF = MMI->getMachineFunction(*F);
+    if (!MF)
+      continue;
+    MachineRegisterInfo &MRI = MF->getRegInfo();
+    //MachineIRBuilder MIRBuilder(*MF);
```
Comment (MrSidims) [inline llvm/lib/Target/SPIRV/SPIRVModuleAnalysis.cpp:1621]: typo

File: llvm/test/CodeGen/SPIRV/extensions/SPV_INTEL_optnone.ll [hunk 2]
```diff
@@ -1,25 +1,20 @@
...
-; RUN: llc -verify-machineinstrs -O0 -mtriple=spirv32-unknown-unknown --spirv-ext=+SPV_INTEL_optnone %s -o - | FileCheck %s --check-prefixes=CHECK,CHECK-EXTENSION
-; RUN: llc -verify-machineinstrs -O0 -mtriple=spirv32-unknown-unknown %s -o - | FileCheck %s --check-prefixes=CHECK,CHECK-NO-EXTENSION
+; RUN: llc -verify-machineinstrs -O0 -mtriple=spirv32-unknown-unknown --spirv-ext=+SPV_INTEL_optnone %s -o - | FileCheck %s --check-prefixes=CHECK-EXTENSION
+; RUN: llc -verify-machineinstrs -O0 -mtriple=spirv32-unknown-unknown %s -o - | FileCheck %s --check-prefixes=CHECK-NO-EXTENSION
 
-; CHECK-EXTENSION: OpCapability OptNoneINTEL
+; RUN: %if spirv-tools %{ llc -O0 -mtriple=spirv64-unknown-unknown --spirv-ext=+SPV_INTEL_optnone %s -o - -filetype=obj | spirv-val %}
+; RUN: %if spirv-tools %{ llc -O0 -mtriple=spirv64-unknown-unknown %s -o - -filetype=obj | spirv-val %}
+
+; CHECK-EXTENSION: OpCapability OptNoneEXT
```
Comment (MrSidims) [inline llvm/test/CodeGen/SPIRV/extensions/SPV_INTEL_optnone.ll:7]: I'm fine with EXT spelling for the text format.

**Review by MrSidims — APPROVED** [review]
SPV_EXT_optnone changes are lgtm

**Review by michalpaszkowski — APPROVED** [review]

**Pull request merged**